include/dxc/Support/WinAdapter.h

```diff
@@ -105,6 +105,9 @@
 #define ERROR_IO_DEVICE EIO
 #define ERROR_INVALID_HANDLE EBADF
 
+// Error returned explicitly. Windows value used.
+#define ERROR_DLL_INIT_FAILED 0x45A
```

Comment:
Since we use std::strerror for displaying errors on Linux, we should use a [valid errno](http://www-numi.fnal.gov/offline_software/srt_public_context/WebDocs/Errors/unix_system_errors.html)
https://github.com/google/DirectXShaderCompiler/blob/bd6553a779750dd42b2e681440809cf16f7d0b09/lib/DxcSupport/dxcapi.use.cpp#L42-L44

I suggest using either ELIBACC or ELIBBAD

Reply:
Well... I guess if we want to return a valid errno, we can't pass it to HRESULT_FROM_WIN32...

Reply:
ughh... since the return type is HRESULT, perhaps we can't use an errno after all...
Should we at least call `SetLastError(ELIBACC)` ?

Reply:
What I would do is manually remove the .so file and run dxc and see what the behavior is...
It should not crash, and ideally it should provide some sort of meaningful message.

Reply:
I can't remember off the top of my head what the stack looks like when handling this error. Depending on how the return value is handled, either returning the errno or HRESULT_FROM_WIN32(errno) or SetLastError(errno) should do it.